plugin/kubernetes/object/pod.go

```diff
@@ -0,0 +1,70 @@
+package object
+
+import (
+	api "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// Pod is a stripped down api.Pod with only the items we need for CoreDNS.
+type Pod struct {
+	Version   string
+	PodIP     string
+	Name      string
+	Namespace string
+	Deleting  bool
+
+	*Empty
+}
+
+// ToPod converts an api.Pod to a *Pod.
+func ToPod(obj interface{}) interface{} {
+	pod, ok := obj.(*api.Pod)
+	if !ok {
+		return nil
+	}
+
+	p := &Pod{
+		Version:   pod.GetResourceVersion(),
+		PodIP:     pod.Status.PodIP,
+		Namespace: pod.GetNamespace(),
+		Name:      pod.GetName(),
+	}
+	t := pod.ObjectMeta.DeletionTimestamp
```

Comment:
Is it worth moving this check above where we define the pod ( line 26 ) and skip adding the pod if it's terminating?
Or rather, do we actually need to store the pod information if it's in a deleting state?